src/main/scala/scala/tools/refactoring/common/TreeTraverser.scala

```diff
@@ -412,5 +421,47 @@ trait TreeTraverser extends TracingImpl {
           super.traverse(t)
       }
     }
+
+    /*
+     * Refined types need special handling, since their original representation might
+     * lack symbols. This method takes care of this.
+     */
+    private def handleRefinedType(orig: Tree, tpe: RefinedType): Unit = {
+      orig.tryMatch {
+        case orig: RefTree if orig.symbol == NoSymbol =>
+          tpe.parents.foreach(handleParentTypeInRefinedType(_, orig))
+
```

Comment:
Indeed, you spotted a weakness here, not even
```scala
trait A
trait B

trait T { this: A with B =>

}
```
works. Thanks for spotting, I will look into this!